venv/Lib/site-packages/pip/_internal/metadata/base.py

```
@@ -0,0 +1,688 @@
+        :raise NoneMetadataError: If ``path`` exists in the info directory, but
+            cannot be read.
+        """
+        raise NotImplementedError()
+
+    def iter_entry_points(self) -> Iterable[BaseEntryPoint]:
+        raise NotImplementedError()
+
+    def _metadata_impl(self) -> email.message.Message:
+        raise NotImplementedError()
+
+    @functools.lru_cache(maxsize=1)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)